### DISCUSSION TITLE
Any hope to get r8152.ko compiled and available as an option ?

### DISCUSSION THREAD

**Discussion by thaneb** [gnuton/asuswrt-merlin.ng#699] [Ideas]
Hi there, Sir Gnuton 🫡😁🙏🏻
Is there any chance we could have a compiled r8152.ko driver along your builds?
I'm using the TUF-AX3000, and thanks to your work, it's far from obsolete, really appreciate your commitment.
I was hoping to improve transfers rate to and from my NAS (2nd hand hp elitedesk 800 G4 with a usb-c 2.5Gbe adapter) with a usb-a 2.5Gbe adapter plugged inthe USB 3 port of the tuf-AX3000...  got it handled by the driver cdc_ncm exactly as described here:
https://wu.renjie.im/blog/network/ax88u-nbase-t/
Ready to be used as wanport or claiming to be lan port after unloading cdc_ncm driver.. but it seems the rtl8152.ko is still missing, and it's far from easy to follow instructions gave here or there.
Same here
https://www.snbforums.com/threads/need-help-to-compile-realtek-r8152-driver-for-rt-ax88u.89035/

It would be much more easier and comfortable if you could include this driver in your build process 🥹

**Comment by gnuton** [comment]
HEy @thaneb I do not have any time to play with this,
but you can leverage the cloud and done some changes and let github build it for you.

**Comment by thaneb** [comment]
ok, so, I'm a total noob sorry, i asked support from the chatgpt (sometimes i am not sure it really helps..) Here's what i did:

I forked your repo
in WLS 2, i pulled my fork

`# release/src-rt/router/rtl8152/`
added rtl8152 source folder in release/src-rt/router/ 

`# in the rtl8152 folder, as i read "replace strscpy function since old kernel 4.x doesn't support this function."`
sed -i -e 's/strscpy/strncpy/g' r8152.c

`# to include rtl8152 in the building process`
echo "obj-m += rtl8152/" >> release/src-rt/router/Makefile 

echo "Adding rtl8152 source for GitHub workflow building" > rtl8152_integration.md

git add .
git commit -m "Add rtl8152 module and update Makefile"
git push origin master

then i went to github action tab on the fork repo, 1 workflow has run, but with 14 errors:

This request has been automatically failed because it uses a deprecated version of `actions/upload-artifact: v2`. Learn more: https://github.blog/changelog/2024-02-13-deprecation-notice-v1-and-v2-of-the-artifact-actions/

can you please confirm that these steps suggested by chatgpt are correct before I start to dig the errors ?

**Reply by gnuton** [reply]
the deprecated artifacts should not stop you to build your pipeline. you should just get warnings. 
try to see if it works for a branch without modifications

**Comment by gnuton** [comment]
To enable such module you should rather uncomment the correct lines
see
https://github.com/gnuton/asuswrt-merlin.ng/commit/47eb98ab31f8e080cb6fe6da7608e7ddcd16c31a

you have to find where the linux configuration is for your router (see target.mak file)

**Comment by thaneb** [comment]
ok, so i deleted my first fork repo, fork yours again, changing the 3

release/src-rt-5.02axhnd.675x/kernel/linux-4.1/config_base.6a.47622
release/src-rt-5.02axhnd.675x/kernel/linux-4.1/config_base.6a.63178
release/src-rt-5.02axhnd.675x/kernel/linux-4.1/config_base.6a.6750

`# CONFIG_USB_RTL8152 is not set` 
to 
`CONFIG_USB_RTL8152=m`

autobuild failed.

delete the repo, fork again, changing only the title to "GNUton's builds with rtl8152 as kernel module"
autobuild failed.

always same 14 errors :
`Error: This request has been automatically failed because it uses a deprecated version of `actions/upload-artifact: v2`. Learn more: https://github.blog/changelog/2024-02-13-deprecation-notice-v1-and-v2-of-the-artifact-actions/`

**Comment by thaneb** [comment]
Ok, line 164 of github-actions.yml
uses: actions/upload-artifact@v2
Changed to 
uses: actions/upload-artifact@v3.1.3
And it seems to work

**Reply by mdreamfly** [reply]
Hello, have you successfully compiled? Can you share your driver files?

**Reply by thaneb** [reply]
hi, yes I did,  unfortunately, the performance is.. disappointing  

https://github.com/thaneb/asuswrt-merlin.ng/releases/tag/r8152.ko_driver-3004.388.8_2-gnuton1

the driver works, and the link is usable as lan, but the perf sucks compare to the integrated gigabit port.
it seems the setting or driver only relies on CPU cores whereas integrated ports or even the adapter when used as wan don't.
I can publish the whole firmware including the driver if you need. If you find a way to make it reach an acceptable speed, please share the solution!

By the way, the fork you did from my fork should auto-compile, you only need to go to its "action" tab, activate, and modify 1 character in the read me, and the auto-build will start.
